### PR TITLE
Add Changesets to automate releases

### PR DESCRIPTION
**Pull request by Jordan-Mysten**
This adds [changesets](https://github.com/changesets/changesets) to automate the release process. When making a PR, you can run `pnpm changeset` and document the changes you'll be making. An action will then open a PR that can be merged to initiate the release process.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
         "build:ts": "tsc --build --verbose tsconfig.all.json",
         "build:other": "pnpm --recursive --workspace-concurrency=0 run build",
         "package": "pnpm --recursive --workspace-concurrency=0 run package",
-        "release": "pnpm run clean && pnpm run build:ts && pnpm run package && pnpm run build:other",
+        "release": "pnpm run clean && pnpm run build:ts && pnpm run package && pnpm run build:other && changeset publish && git push --follow-tags && git status",
```
Comment (jordansexton) [inline package.json:18]: I wonder if we should add another command here, "release" was supposed to mean "(build for) release"

Comment (Jordan-Mysten) [reply package.json:18]: Isn't `build` just build for release?

Comment (jordansexton) [reply package.json:18]: `build` doesn't do a fresh build or shim the package.json files. I think we can just rename the current command to `build:release`, and then `release` can call it and do the other stuff

Comment (Jordan-Mysten) [reply package.json:18]: I'll do this, although IMO building for release should really just be `build`. If there's additional work to be done before publishing it really should be in `prepare` or `prepublishOnly`, but we can sort that out later.

**Pull request merged**